R/as_draws_df.R

```diff
@@ -90,47 +59,7 @@ as_draws_df.draws_array <- function(x, ...) {
 
 #' @rdname draws_df
 #' @export
-as_draws_df.draws_list <- function(x, ...) {
-  if (ndraws(x) == 0L) {
-    return(empty_draws_df(variables(x)))
-  }
-  iteration_ids <- iteration_ids(x)
-  chain_ids <- chain_ids(x)
-  vars <- names(x[[1L]])
-  x <- do.call(rbind.data.frame, x)
-  colnames(x) <- vars
-  x <- tibble::as_tibble(x)
-  x[".chain"] <- rep(chain_ids, each = max(iteration_ids))
-  x[".iteration"] <- rep(iteration_ids, max(chain_ids))
-  x[".draw"] <- seq_len(nrow(x))
-  class(x) <- class_draws_df()
-  x
-}
-
-#' @rdname draws_df
-#' @export
-as_draws_df.draws_rvars <- function(x, ...) {
-  as_draws_df(as_draws_array(x), ...)
-}
-
-#' @rdname draws_df
-#' @export
-as_draws_df.mcmc <- function(x, ...) {
-  as_draws_df(as_draws_matrix(x), ...)
-}
-
-#' @rdname draws_df
-#' @export
-as_draws_df.mcmc.list <- function(x, ...) {
-  as_draws_df(as_draws_array(x), ...)
-}
-
-#' Convert any \R object into a `draws_df` object
-#' @param x An \R object.
-#' @param .iteration optional name of the column containing iteration indices
-#' @param .chain optional name of the column containing chain indices
-#' @noRd
-.as_draws_df <- function(x, .iteration = NULL, .chain = NULL) {
+as_draws_df.data.frame <- function(x, .iteration = NULL, .chain = NULL, ...) {
```

Comment:
Just a head-up; I realized I didn't pay much attention to ellipsis (...) in this PR. Depending on your answer to #158, I'll go through the code and make sure they are placed when necessary.